### PR TITLE
Add module to extract message values for use in scripting

### PR DESCRIPTION
**Pull request by bhannawa**
This commit creates a module `extract_message` for extracting contents from a ULog message into a list to be used for scripting with ULog files. The module contains one function that takes in a ULog file, the desired message, and optionally a start and end time and returns a list of records from the ULog message as key-value pairs. I created this code by modifying the ulog2csv module.

### CODE REVIEW

File: pyulog/extract_message.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    if not type(message) == str:
+        raise AttributeError("Must provide a message to pull from ULog file")
+
+    msg_filter = message.split(',') if message else None
+
+    ulog = ULog(ulog_file_name, msg_filter, disable_str_exceptions)
+    data = ulog.data_list
+
+    if not data:
+        raise AttributeError("Provided message is not in the ULog file")
```
Comment (bkueng) [inline pyulog/extract_message.py:30]: Can you use `ulog.get_dataset(message)` here?

**Review by bkueng — COMMENTED** [review]
Generally looks good.

Can you add some unit test under https://github.com/PX4/pyulog/tree/main/test?

And can you check the relevant CI failures? https://github.com/PX4/pyulog/actions/runs/11639019887/job/32414624361?pr=106
```
pyulog/extract_message.py:8:0: C0301: Line too long (112/100) (line-too-long)
pyulog/extract_message.py:21:11: C0123: Use isinstance() rather than type() for a typecheck. (unidiomatic-typecheck)
pyulog/extract_message.py:49:12: C0200: Consider using enumerate instead of iterating with range and len (consider-using-enumerate)
```

File: pyulog/extract_message.py (hunk 1 above)
Comment (bhannawa) [reply pyulog/extract_message.py:30]: Yes I can. Looks like that method raises an IndexError if the message provided does not appear in the file, so I will raise an additional AttributeError in this extract_message module to be more descriptive about what caused the error.

File: pyulog/extract_message.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+    :return: (list[dict]) A list of each record from the ULog as key-value pairs
+    """
+
+    if not type(message) == str:
+        raise AttributeError("Must provide a message to pull from ULog file")
+
+    msg_filter = message.split(',') if message else None
+
+    ulog = ULog(ulog_file_name, msg_filter, disable_str_exceptions)
```
[the remark this answers is not in the document]
Comment (bhannawa) [reply pyulog/extract_message.py:26]: Good catch. Was hoping to figure out a way to handle multiple messages at once, but that isn't needed for my use case and it seems simpler to make a list of the messages and iterate through that instead.

**Comment by bhannawa** [comment]
Okay thanks for your input. I was able to implement the changes you suggested (ulog.data_list -> ulog.get_dataset(message), removing the messages.split) and resolve all the CI failures (related to this extract_message module). I did create a very basic unit test by modifying the ulog2csv test

**Review by bkueng — APPROVED** [review]
Looks good, thanks

**Pull request merged**